java/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.java

```diff
@@ -55,176 +57,185 @@
-
-    // add graphics overlays to the map view
-    mMapView.getGraphicsOverlays().add(mResultGraphicsOverlay);
-    mMapView.getGraphicsOverlays().add(mInputGraphicsOverlay);
-
-    mGeoprocessingTask = new GeoprocessingTask(getString(R.string.viewshed_service));
-
-    mMapView.setOnTouchListener(new DefaultMapViewOnTouchListener(getApplicationContext(), mMapView) {
-      @Override public boolean onSingleTapConfirmed(MotionEvent e) {
-        android.graphics.Point screenPoint = new android.graphics.Point(Math.round(e.getX()), Math.round(e.getY()));
-        Point mapPoint = mMapView.screenToLocation(screenPoint);
-        addGraphicForPoint(mapPoint);
-        calculateViewshedAt(mapPoint);
-        return super.onSingleTapConfirmed(e);
-      }
-    });
-  }
-
-  /**
-   * Adds a graphic at the chosen mapPoint.
-   *
-   * @param point in MapView coordinates.
-   */
-  private void addGraphicForPoint(Point point) {
-    // remove existing graphics
-    mInputGraphicsOverlay.getGraphics().clear();
-
-    // new graphic
-    Graphic graphic = new Graphic(point);
-
-    // add new graphic to the graphics overlay
-    mInputGraphicsOverlay.getGraphics().add(graphic);
-  }
-
-  /**
-   * Uses the given point to create a FeatureCollectionTable which is passed to performGeoprocessing.
-   *
-   * @param point in MapView coordinates.
-   */
-  private void calculateViewshedAt(Point point) {
-    // remove previous graphics
-    mResultGraphicsOverlay.getGraphics().clear();
-
-    // cancel any previous job
-    if (mGeoprocessingJob != null) {
-      mGeoprocessingJob.cancel();
+    private MapView mMapView;
+    private GeoprocessingTask mGeoprocessingTask;
+    private GeoprocessingJob mGeoprocessingJob;
+
+    private GraphicsOverlay mInputGraphicsOverlay;
+    private GraphicsOverlay mResultGraphicsOverlay;
+    // objects that implement Loadable must be class fields to prevent being garbage collected before loading
+    private FeatureCollectionTable mFeatureCollectionTable;
+    private View mLoadingView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY);
+
+        mInputGraphicsOverlay = new GraphicsOverlay();
+        mResultGraphicsOverlay = new GraphicsOverlay();
+
+        // inflate MapView and the LoadingView from layout
+        mMapView = findViewById(R.id.mapView);
+        mLoadingView = findViewById(R.id.loadingView);
+        // create a map with a topographic basemap
+        ArcGISMap map = new ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC);
+        // set the map to be displayed in this view
+        mMapView.setMap(map);
+        mMapView.setViewpoint(new Viewpoint(45.3790902612337, 6.84905317262762, 100000));
```

Comment:
```suggestion
        mMapView.setViewpoint(new Viewpoint(45.3790, 6.8490, 100000));
```

Point doesn't need to be so precise

Reply:
I've changed this in both samples.